src/v/cluster/controller_backend.cc

```diff
@@ -740,7 +743,45 @@ controller_backend::force_replica_set_update(
               last_cloud_offset.error());
             co_return last_cloud_offset.error();
         }
-        if (last_cloud_offset.value() > partition->dirty_offset()) {
+
+        vlog(
+          clusterlog.info,
+          "[{}] force-update replica set - last cloud offset {}, dirty offset: "
+          "{}",
+          partition->ntp(),
+          last_cloud_offset.value(),
+          partition->dirty_offset());
+
+        /**
+         * This variable indicates whether the partition can be recovered
+         * by the leader.
+         *
+         * When force reconfiguring partition controller backed decides which of
+         * the new replicas should be added to the partition configuration as
+         * learners. The logic here is relatively simple: if a replica is not
+         * already in the replica set i.e. it is the disaster survivor it is
+         * added to the replica set as a voter serving as a source of truth.
+         * Nodes that joins the replica set are added to the configuration as
+         * learners in order to prevent them from reaching a majority and
+         * overcoming the current minority (the survivors).
+         *
+         * After establishing which replicas are learners vs voters the
+         * condition belows decides if the survivor should be treated as a
+         * source of truth or if cloud data contains more up to date state.
+         *
+         * If replica dirty offset is greater than last cloud offset it means
+         * that the replica is more up to date, otherwise data in the bucket are
+         * newer and they should be used. This condition should only be verified
+         * if a node is not a learner and can not be recovered from the cloud.
+         * Otherwise all learners would face multiple deletions while being
+         * recovered by the leader as their dirty offset may smaller than last
+         * cloud offset for a very long time.
+         *
```

Comment:
I would appreciate if you could rephrase some sentences.

I failed to make sense of the first sentence of the second paragraph. I'm not sure what the word `backed` relates to, also the word `when` suggests a complex sentence but I cannot see the main clause.
`if a replica is not already in the replica set i.e. it is the disaster survivor ...` -- why is this equivalent?
`Nodes that joins` -> `join`?
`This condition should only be verified if a node is not a learner and can not be recovered from the cloud` -- from the _cloud_? I thought from the leader.
`may smaller` -> `may BE smaller`?

Reply:
You are right, i changed the wording in the paragraph. As for the second part: the condition determines if the leaders offset is smaller